File: app/src/main/java/com/timetracker/db/DbHelper.java
package com.timetracker.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.timetracker.db.ActionsContract.ActionEntry;

public class DbHelper extends SQLiteOpenHelper {

    public static final int DATABASE_VERSION = 1;
    public static final String DATABASE_NAME = "Records.db";

    public DbHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(SQL_CREATE_ACTIONS);
        db.execSQL(SQL_CREATE_CATEGORIES);
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }

    private static final String SQL_CREATE_ACTIONS =
            "CREATE TABLE " + ActionEntry.TABLE_NAME + " (" +
                    ActionEntry._ID + " INTEGER PRIMARY KEY," +
                    ActionEntry.COLUMN_NAME_DATE + " INTEGER," +
                    ActionEntry.COLUMN_NAME_TYPE + " TEXT," +
                    ActionEntry.COLUMN_NAME_CATEGORY_ID + " INTEGER)";

    private static final String SQL_CREATE_CATEGORIES =
            "CREATE TABLE " + CategoriesContract.CategoryEntry.TABLE_NAME + " (" +
                    CategoriesContract.CategoryEntry._ID + " INTEGER PRIMARY KEY," +
                    CategoriesContract.CategoryEntry.COLUMN_NAME_NAME + " TEXT)";

}

File: app/src/main/java/com/timetracker/db/DbHelper.kt
package com.timetracker.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.provider.BaseColumns

import com.timetracker.db.entries.*

class DbHelper(context: Context) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {

    override fun onCreate(db: SQLiteDatabase) = tableCreators.forEach { db.execSQL(it) }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        if (newVersion == 2) {
            db.execSQL("""CREATE TABLE ${GoalEntry.TABLE_NAME} (
                        ${BaseColumns._ID} INTEGER PRIMARY KEY,
                        ${GoalEntry.COLUMN_CATEGORY_ID} INTEGER NOT NULL,
                        ${GoalEntry.COLUMN_TYPE} TEXT NOT NULL,
                        ${GoalEntry.COLUMN_DURATION} INTEGER NOT NULL)""")
        }
    }

    companion object {

        val DATABASE_VERSION = 2
        val DATABASE_NAME = "Records.db"

        private val tableCreators = listOf(
                """CREATE TABLE ${ActionEntry.TABLE_NAME} (
                        ${BaseColumns._ID} INTEGER PRIMARY KEY,
                        ${ActionEntry.COLUMN_NAME_DATE} INTEGER,
                        ${ActionEntry.COLUMN_NAME_TYPE} TEXT,
                        ${ActionEntry.COLUMN_NAME_CATEGORY_ID} INTEGER)""",

                """CREATE TABLE ${CategoryEntry.TABLE_NAME} (
                        ${BaseColumns._ID} INTEGER PRIMARY KEY,
                        ${CategoryEntry.COLUMN_NAME_NAME} TEXT)""",

                """CREATE TABLE ${GoalEntry.TABLE_NAME} (
                        ${BaseColumns._ID} INTEGER PRIMARY KEY,
                        ${GoalEntry.COLUMN_CATEGORY_ID} INTEGER NOT NULL,
                        ${GoalEntry.COLUMN_TYPE} TEXT NOT NULL,
                        ${GoalEntry.COLUMN_DURATION} INTEGER NOT NULL)""")
    }

}
